src/writer.rs

```diff
@@ -394,4 +417,39 @@ mod indentation {
                 .as_ref()
         );
     }
+    #[test]
+    fn node_writer_empty() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_node(b"empty")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
```

Comment:
What do you think about passive ```with_attribute``` vs active ```push_attribute```?

I'm conflicted because then it uses active ```write_empty```.  There's decent arguments for both options

Reply:
I tend to use `with_...` for a builder like pattern (something that returns a `&mut Self`) and `push_...` for something that return `()`. It is more a convention thing but in your case with_attribute is better.